### ISSUE TITLE
Specifying go-version 1.19 defaults to 1.19.2, as opposed to the actual major version 1.19

### ISSUE THREAD

**Issue by sebastian-quintero** [actions/setup-go#285]
**Description:**
I have specified an action with the specific version `1.19`. I want that specific major version. However, the action is getting `1.19.2`, which is not what I want. You can see a screenshot here.

<img width="1053" alt="image" src="https://user-images.githubusercontent.com/42215971/199359127-bf0d46f7-c034-4e2f-808e-0c09c89002b5.png">

**Action version:**
v3

**Platform:**
- [x] Ubuntu
- [x] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
1.19

**Repro steps:**  
Set up an action with the following characteristics:

```yml
 - name: set up go 1.19
    uses: actions/setup-go@v3
    with:
      go-version: 1.19
      check-latest: false
      cache: false
    id: go
```

**Expected behavior:**
I expect the action to configure `1.19`, not `1.19.2`.

**Actual behavior:**
The action is configuring `1.19.2`.

**Comment by e-korolevskii** [comment]
Hello @sebastian-quintero 

We will investigate this issue and get back to you soon.

**Comment by e-korolevskii** [comment]
Hello @sebastian-quintero 

In this situation, this is expected behavior. As stated in the [documentation](https://github.com/actions/setup-go#v3), we are using the [semver spec](https://github.com/npm/node-semver). Therefore, specifying version `1.19` is recognized as `1.19.X` ([more details](https://github.com/npm/node-semver#x-ranges-12x-1x-12-)). To get the exact `1.19` version, you should specify the full version notation `1.19.0`.

Now I'm closing this issue, but if you have any questions, you can ask them here or open a separate one! 
Have a good day!

**Comment by jadn123** [comment]
I interpreted the issue as `1.19.2` was pulled instead of the latest `1.19.3`. At least that's the issue I was facing.

**Comment by dmitry-shibanov** [comment]
Hello @jadn123. Could you please try to set check-latest: true. The action firstly will check the hostedToolCache directory. If the version is not found in the hostedToolCache the action will try to download it from go-versions.  We updated versions-manifest.json that is why it should take the latest version.

**Comment by sebastian-quintero** [comment]
@e-korolevskii setting it as `1.19.0` solved the issue. 🌮 for you! Have a good day.

**Comment by jadn123** [comment]
@dmitry-shibanov 
You are right. Adding `check-latest: true` worked now 👍
So I assume it was dependent on this one https://github.com/actions/go-versions/pull/63 ?

```
Setup go version spec 1.19
Attempting to resolve the latest version from the manifest...
matching 1.19...
Resolved as '1.19.3'
Attempting to download 1.19.3...
matching 1.19.3...
Acquiring 1.19.3 from https://github.com/actions/go-versions/releases/download/1.19.3-3375666673/go-1.19.3-linux-x64.tar.gz
Extracting Go...
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/xyz -f /home/runner/work/_temp/xyz
Successfully extracted go to /home/runner/work/_temp/xyz
Adding to the cache ...
Successfully cached go to /opt/hostedtoolcache/go/1.[19].3/x64
Added go to the path
Successfully set up Go version 1.19
go version go1.19.3 linux/amd64
```

**Comment by li-jin-gou** [comment]
> Hello @jadn123. Could you please try to set check-latest: true. The action firstly will check the hostedToolCache directory. If the version is not found in the hostedToolCache the action will try to download it from go-versions. We updated versions-manifest.json that is why it should take the latest version.

I set check-latest: true, but it still installs version 1.19.2 instead of 1.19.3. this is pr https://github.com/gofiber/fiber/pull/2197.  ❤️
